tools/distpartitioning/data_shuffle.py

```diff
@@ -12,82 +12,113 @@
 from datetime import timedelta
 from dataset_utils import get_dataset
 from utils import read_partitions_file, read_json, get_node_types, \
-                    augment_node_data, augment_edge_data, get_ntypes_map, \
+                    augment_edge_data, get_gnid_range_map, \
                     write_dgl_objects, write_metadata_json
 from gloo_wrapper import alltoall_cpu_object_lst, alltoallv_cpu, \
                     alltoall_cpu, allgather_sizes, gather_metadata_json
 from globalids import assign_shuffle_global_nids_nodes, \
-                    assign_shuffle_global_nids_edges, get_shuffle_global_nids_edges
-from convert_partition import create_dgl_object, create_metadata_json, validateDGLObjects
+                    assign_shuffle_global_nids_edges, \
+                    get_shuffle_global_nids_edges
+from convert_partition import create_dgl_object, create_metadata_json
 
-def exchange_node_data(rank, world_size, node_data):
-    """
-    Exchange node_data among the processes in the world
-    Prepare the list of slices targeting each of the process and
-    trigger alltoallv_cpu for the message exchange.
+def gen_node_data(rank, world_size, node_part_ids, ntid_ntype_map, nid_schema_map):
+    '''
+    For this data processing pipeline, reading node files is not needed. All the needed information about
+    the nodes can be found in the metadata json file. This function generates the nodes owned by a given
+    process, using metis partitions. 
 
-    Parameters:
+    Parameters: 
     -----------
     rank : int
-        rank of the current process
+        rank of the process
     world_size : int
-        total no. of participating processes
-    node_data : dictionary
-        nodes data dictionary with keys as column names and values as
-        columns from the nodes csv file
+        total no. of processes
+    node_part_ids : 
+        numpy array, whose length is same as no. of nodes in the graph. Index in this array is the global_nid
+        and value is the partition-id which owns the node
+        TODO: For large graphs, this mapping function can be memory intensive. This needs to be changed to 
+        processes owning a set of global-nids, per partitioning algorithm, and messaging will be used to 
+        identify the ownership instead of mem. lookups. 
```

Comment:
please don't add TODO in the docstrng like this. add the TODO as a comment inside the function.

Reply:
Moved the TODO to the gen_dist_partitions function, where we read the nodeid<-> partition-ids for the first time in the pipeline.